lib/blockscore/base.rb

```diff
@@ -53,22 +52,22 @@ def self.resource
       @resource ||= Util.to_underscore(to_s.split('::').last)
     end
 
-    def self.api_url
-      'https://api.blockscore.com/'
-    end
-
     def self.endpoint
       fail NotImplementedError, ABSTRACT_WARNING if equal?(Base)
 
-      "#{api_url}#{Util.to_plural(resource)}"
+      API_URL + Util.to_plural(resource)
     end
 
     def persisted?
-      !id.nil? && !attributes[:deleted]
+      !id.nil? && !attributes.fetch(:deleted, false)
     end
 
     protected
```

Comment:
What is `protected` being used for in this case? In ruby `protected` is usually not what people think it is.

All it does is provides an API that other objects of the same type can use to access internal state. It's useful in things like `#eql?` or `#==` where you want to compare against some other object but the public API doesn't give you access to the internal state you need.

Most of the time when people use `protected` in ruby they have a different understanding of what it's doing, and it is probably a bug except when it satisfies a very narrow use case.